### PR TITLE
Udt datatype tests

### PR DESCRIPTION
**Pull request by kishkaru**
Work in progress for UDT tests.

### CODE REVIEW

File: driver-core/src/test/java/com/datastax/driver/core/UserTypesTest.java [hunk 1]
```diff
@@ -38,29 +46,208 @@
...
+            // insert UDT data
+            UserType alldatatypesDef = cluster.getMetadata().getKeyspace("testPrimitiveDatatypes").getUserType("alldatatypes");
+            UDTValue alldatatypes = alldatatypesDef.newValue();
+
+            int startIndex = (int) 'a';
+            for (int i = 0; i < DATA_TYPE_PRIMITIVES.length(); i++) {
+                Datatype datatype = DATA_TYPE_PRIMITIVES.get(i);
+                switch (dataType.getName()) {
+                    case ASCII:
+                        alldatatypes.setString(Character.toString((char) startIndex + i), SAMPLE_DATA.get(datatype));
```
Comment (joaquincasares) [inline driver-core/src/test/java/com/datastax/driver/core/UserTypesTest.java]: I'd deal with these two parameters 2 lines earlier to remove repeated code.

File: driver-core/src/test/java/com/datastax/driver/core/UserTypesTest.java [hunk 2]
```diff
@@ -38,29 +46,208 @@
...
-            UserType addrDef = cluster.getMetadata().getKeyspace("ks").getUserType("address");
-            UserType phoneDef = cluster.getMetadata().getKeyspace("ks").getUserType("phone");
+            int startIndex = (int) 'a';
+            for (int i = 0; i < DATA_TYPE_NON_PRIMITIVE_NAMES.length(); i++)
+                for (int j = 0; i < DATA_TYPE_PRIMITIVES.length(); j++) {
+                    Datatype.Name name = DATA_TYPE_NON_PRIMITIVE_NAMES.get(i);
+                    Datatype datatype = DATA_TYPE_PRIMITIVES.get(j);
+                    switch(name) {
+                        case LIST:
+                            alldatatypes.setList(Character.toString((char) startIndex + i) + "_" +
```
Comment (joaquincasares) [inline driver-core/src/test/java/com/datastax/driver/core/UserTypesTest.java]: Same comment as above for easier readability.

File: driver-core/src/test/java/com/datastax/driver/core/DataTypeIntegrationTest.java [hunk 3]
```diff
@@ -169,9 +171,32 @@
...
+            List<Object> theList = new ArrayList<Object>();
+            theList.add(SAMPLE_DATA.get(dataType));
+            theList.add(SAMPLE_DATA.get(dataType));
+            return theList;
+        }
+        else if(collectionType == DataType.Name.SET)
+            return ImmutableSet.of(SAMPLE_DATA.get(dataType));
+        else if(collectionType == DataType.Name.MAP)
+            if(dataType.getName() == DataType.Name.BLOB)
+                return new HashMap<Object, Object>().put(SAMPLE_DATA.get(DataType.ascii()), SAMPLE_DATA.get(dataType));
```
Comment (joaquincasares) [inline driver-core/src/test/java/com/datastax/driver/core/DataTypeIntegrationTest.java]: Let's remove this now that we know this is not needed for the java tests.

File: driver-core/src/test/java/com/datastax/driver/core/UserTypesTest.java [hunk 4]
```diff
@@ -66,4 +78,185 @@ public void simpleWriteReadTest() throws Exception {
...
+
+            // retrieve and verify data
+            ResultSet rs = session.execute("SELECT * FROM mytable");
+            List<Row> rows = rs.all();
+            assertEquals(1, rows.size());
+
+            Row row = rows.get(0);
+
+            assertEquals(row.getInt("a"), 0);
+            assertEquals(row.getUDTValue("alldatatypes"), alldatatypes);
```
Comment (olim7t) [inline driver-core/src/test/java/com/datastax/driver/core/UserTypesTest.java]: Should be `row.getUDTValue("b")` (you want the name of the column here, not the name of the type).

Comment (kishkaru) [inline driver-core/src/test/java/com/datastax/driver/core/UserTypesTest.java]: I see my mistake now. I used different nomenclature when I created the table.

**Comment by olim7t** [comment]
Merged in 2.1.

I've merged the 3 pull requests #206, #207 and #208 (which is why they're marked as rejected) and squashed a few commits.

**Pull request closed without merging**